File: src/main/java/su/nsk/iae/post/dsm/manager/ServerUtils.java
package su.nsk.iae.post.dsm.manager;

import java.io.IOException;
import java.net.ServerSocket;

public class ServerUtils {

    public static int findFreePort() {
        try (ServerSocket socket = new ServerSocket(0)) {
            socket.setReuseAddress(true);
            return socket.getLocalPort();
        } catch (IOException e) {
            DSMManagerLogger.error(ServerUtils.class, e.getMessage());
        }
        throw new IllegalStateException("Could not find a free TCP/IP port");
    }
}

File: post-ide-dsm-manager/src/main/kotlin/su/nsk/iae/post/dsm/manager/infrastructure/ServerUtils.kt
package su.nsk.iae.post.dsm.manager.infrastructure

import java.io.IOException
import java.net.ServerSocket

object ServerUtils {

    fun findFreePort(): Int {
        try {
            ServerSocket(0).use { socket ->
                socket.reuseAddress = true
                return socket.localPort
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }
        throw IllegalStateException("Could not find a free TCP/IP port")
    }
}